cpu/sam0_common/periph/uart.c

```diff
@@ -99,18 +113,26 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     if ((rx_cb) && (uart_config[uart].rx_pin != GPIO_UNDEF)) {
         uart_ctx[uart].rx_cb = rx_cb;
         uart_ctx[uart].arg = arg;
-#if defined (CPU_SAML1X) || defined (CPU_SAMD5X)
+#ifdef UART_HAS_TX_ISR
+        /* enable RXNE ISR */
         NVIC_EnableIRQ(SERCOM0_2_IRQn + (sercom_id(dev(uart)) * 4));
 #else
+        /* enable UART ISR */
         NVIC_EnableIRQ(SERCOM0_IRQn + sercom_id(dev(uart)));
```

Comment:
Got a hard time debugging this one.
In the current shape the provided test app doesn't work at all on SAML21.
No DRE interrupt is triggered. In fact, the NVIC_EnableIRQ is never call because their is no callback provided so we did not enter [here](https://github.com/RIOT-OS/RIOT/blob/7a16375438e960d46cecddae7e823c879d56a087/cpu/sam0_common/periph/uart.c#L113).
The callback is not provided because weren't using a shell.
So we need a second way to enable the NVIC in this specific case (TX only, no RX).

Reply:
Yea you are right.
I only tested the non-callback case on samd5x, for samd2x/saml2x the UART interrupt was only enabled when a callback was provided.
It is now also enabled when `MODULE_PERIPH_UART_NONBLOCKING` is defined.